apisix/core/config_etcd.lua

```diff
@@ -416,16 +416,20 @@ local function _automatic_fetch(premature, self)
         return
     end
 
-    local etcd_cli, _, err = etcd.new(self.etcd_conf)
-    if not etcd_cli then
-        error("failed to start a etcd instance: " .. err)
-    end
-    self.etcd_cli = etcd_cli
-
     local i = 0
     while not exiting() and self.running and i <= 32 do
         i = i + 1
+
         local ok, err = xpcall(function()
+            if not self.etcd_cli then
+                local etcd_cli, err = etcd.new(self.etcd_conf)
+                if not etcd_cli then
+                    error("failed to create etcd instance for key ["
+                          .. self.key .. "]: " .. (err or "unknown"))
+                end
+                self.etcd_cli = etcd_cli
+            end
+
```

Comment:
Why has the number of returned values changed?

Reply:
the old code is wrong. the return value is `res, err` for latest version of `resty-etcd`